### ISSUE TITLE
Style/ClassAndModuleChildren with compact style ignores outer class with ineheritance

### ISSUE THREAD

**Issue by mockdeep** [rubocop/rubocop#9817]
With the `Style/ClassAndModuleChildren` rule and `EnforcedStyle: compact`, it ignores instances where the outer class inherits:

```rb
class Target < ApplicationRecord
  class Refresh
  end
end
```

--------

## Expected behavior

The rule should report a violation. It should require:

```rb
class Target::Refresh
end
```

## Actual behavior

The rule does not report a violation.

## Steps to reproduce the problem

See above.

## RuboCop version

```
1.15.0 (using Parser 3.0.1.1, rubocop-ast 1.5.0, running on ruby 3.0.1 x86_64-linux)
  - rubocop-rails 2.10.1
  - rubocop-rspec 2.3.0
 ```

**Comment by koic** [comment]
I think this is an expected behavior. Probably not enough documentation.
https://github.com/rubocop/rubocop/blob/v1.15.0/spec/rubocop/cop/style/class_and_module_children_spec.rb#L348-L355

**Comment by mockdeep** [comment]
Hmm, is there any way we can resolve both scenarios? Can the file path be used to infer which is the primary class in the file? Like, if the filename is `refresh.rb` then it should be the shorthand, but if it's `target.rb` the longhand?

**Comment by koic** [comment]
Hm, I don't know if there is the issue, so let me check. These bad and good cases are incompatible.

```ruby
# bad?
class Target < ApplicationRecord
  class Refresh
  end
end

# good?
class Target::Refresh
end
```

Should there be different code in the expected good case?

**Comment by koic** [comment]
Due to a lack of activity, I'll close this issue. If the problem continues, please feel free to reopen it with additional information.
